dart/dynamics/Entity.h

```diff
@@ -0,0 +1,168 @@
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ *   CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *   INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ *   MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *   DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ *   CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+ *   USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ *   AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *   LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *   ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *   POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef DART_DYNAMICS_ENTITY_H_
+#define DART_DYNAMICS_ENTITY_H_
+
+#include <Eigen/Core>
+#include <string>
+#include <vector>
+
+#include "dart/dynamics/Shape.h"
+
+namespace dart {
+namespace renderer {
+class RenderInterface;
+} // namespace renderer
+} // namespace dart
+
+namespace dart {
+namespace dynamics {
+
+class Frame;
+class Shape;
+
+/// Entity class is a base class for any objects that exist in the kinematic
+/// tree structure of DART.
+///
+/// Entities all share the following properties: they exist within a reference
+/// frame, have a name, and are visualizable. However, different Entity types
+/// may have different policies about how/if their reference frame or name
+/// can be changed. Use the Detachable class to create an Entity whose reference
+/// Frame can be changed arbitrarily.
+class Entity
+{
+public:
+  friend class Frame;
+
+  /// Constructor for typical usage
+  explicit Entity(Frame* _refFrame, const std::string& _name,
+                  bool _quiet);
+
+  /// Destructor
+  virtual ~Entity();
+
+  /// Set name. Some implementations of Entity may make alterations to the name
+  /// that gets passed in. The final name that this entity will use gets passed
+  /// back in the return of this function.
+  virtual const std::string& setName(const std::string& _name);
+
+  /// Return the name of this Entity
+  virtual const std::string& getName() const;
+
+  /// Add a visualization shape for this Entity
+  virtual void addVisualizationShape(Shape* _p);
+
+  /// Render this Entity
+  virtual void draw(renderer::RenderInterface* _ri = NULL,
+                    const Eigen::Vector4d& _color = Eigen::Vector4d::Ones(),
+                    bool _useDefaultColor = true, int _depth = 0) const;
+
+  /// Get the parent (reference) frame of this Entity
+  Frame* getParentFrame();
+
+  /// Get the parent (reference) frame of this Entity
+  const Frame* getParentFrame() const;
+
+  /// True iff this Entity depends on (i.e. kinematically descends from)
```

Comment:
I use "iff" to mean "if and only if". Is that too confusing? Would it be better to write out "if and only if"?